### PR TITLE
Feat: multi-sample averaging and 50Hz mains filtering

### LINKED ISSUE #18: Missing feature: 50/60Hz noise filtering
The MAX31656 contains a mains frequency notch filter, set to 60Hz by default. This should be changed for applications in 50Hz regions. The arduino library for this breakout board allows for selecting 50Hz:

https://github.com/adafruit/Adafruit_MAX31856/blob/master/Adafruit_MAX31856.cpp#L189

However this feature is missing in the Circuitpython library.

### PR DESCRIPTION
**Pull request by eirinnm**
This PR implements two features from the datasheet:

* On-chip multi-sample averaging (1,2,4,8,16 samples). Default is 1 (no averaging). Using on-chip averaging is much faster than retrieving individual samples and averaging them in user code.
* 50Hz mains frequency notch filter. Default is off (will filter 60Hz instead). #18

### CODE REVIEW

**Comment by ladyada** [comment]
hi please create functions to set/get the freq filter (none, 50, 60?) and oversampling rather than setting once in init :)

**Comment by eirinnm** [comment]
Thanks for the feedback! I've created functions `set_sampling` and `select_mains_filtering` so that these options can be changed after init. For most users they don't have to be called, because the chip has sane defaults (single sampling and 60Hz filter).

**Comment by caternuson** [comment]
Code logic looks generally OK. But instead of actual function functions, CP lib style is to use Python property feature:
https://docs.circuitpython.org/en/latest/docs/design_guide.html#getters-setters

Look at the `temperature_thresholds` property in this library for an example.

With that form, the following syntax is how they would get used in user code to set:
```python
thermocouple.averaging = 16
thermocouple.noise_rejection = True
```
and to query (get):
```python
print("Averaging =", thermocouple.averaging)
print("Noise Rejection =", thermocouple.noise_rejection)
```

In the above example code snippets, `averaging` and `noise_rejection` are the suggested property names, similar to `temperature_thresholds`.

**Comment by eirinnm** [comment]
Yes I see. Thanks for the advice! I've implemented it using property getter/setters now.

File: adafruit_max31856.py [hunk 1]
```diff
@@ -184,16 +240,14 @@ def unpack_temperature(self) -> float:
 
         return temp_float
 
-
-
     @property
     def reference_temperature(self):
         """Wait to retreive temperature of the cold junction in degrees Celsius. (read-only)"""
```
Comment (dhalbert) [inline adafruit_max31856.py]:
```suggestion
        """Wait to retrieve temperature of the cold junction in degrees Celsius. (read-only)"""
```

**Review by dhalbert — CHANGES_REQUESTED** [review]
One spelling fix.

**Comment by caternuson** [comment]
LGTM. Nice job switching to property approach!

Functionally tested, without a TC connected, on Itsy M0:
```python
Adafruit CircuitPython 7.3.2 on 2022-07-20; Adafruit ItsyBitsy M0 Express with samd21g18
>>> import board
>>> import adafruit_max31856
>>> import digitalio
>>> cs = digitalio.DigitalInOut(board.D10)
>>> cs.switch_to_output()
>>> max = adafruit_max31856.MAX31856(board.SPI(), cs)
>>> max.reference_temperature
24.375
>>> max.averaging
1
>>> max.averaging = 16
>>> max.averaging
16
>>> max.averaging = 42
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "adafruit_max31856.py", line 190, in averaging
ValueError: Num_samples must be one of 1,2,4,8,16
>>> max.noise_rejection
60
>>> max.noise_rejection = 50
>>> max.noise_rejection
50
>>> max.noise_rejection = 23
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "adafruit_max31856.py", line 218, in noise_rejection
ValueError: Frequency must be 50 or 60
>>>  
```

**Comment by eirinnm** [comment]
Thanks all, I've enjoyed contributing to this library.

**Review by FoamyGuy — APPROVED** [review]
Looks good to me, verified the open change requested was completed. 

Thanks for the improvement @eirinnm

**Pull request merged**